### PR TITLE
Adding support GridSearchCV and RandomizedGridSearchCV

### LINKED ISSUE #475: Hello! Is it possible to use scikit-learn's Gridsearchcv ?

### PR DESCRIPTION
**Pull request by scnakandala**
Closes #475

### CODE REVIEW

File: tests/test_sklearn_model_selection.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+            model = model_selection_class(base_classifier, parameters, cv=5, **kwargs)
+
+            n_train_rows = int(X.shape[0] * 0.6)
+            model.fit(X[:n_train_rows, :], y[:n_train_rows])
+
+            if not score_w_train_data:
+                X = X[n_train_rows:, :]
+
+            extra_config = {hummingbird.ml.operator_converters.constants.BATCH_SIZE: X.shape[0]}
```
Comment (interesaaat) [inline tests/test_sklearn_model_selection.py]: Why is this needed?

**Review by interesaaat — APPROVED** [review]

File: tests/test_sklearn_model_selection.py (hunk 1 above)
Comment (scnakandala) [reply tests/test_sklearn_model_selection.py]: Agree. It is not needed. So removed it in the latest commit.

File: hummingbird/ml/supported.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 GaussianNB,
 GradientBoostingClassifier,
 GradientBoostingRegressor,
+GridSearchCV,
```
Comment (interesaaat) [inline hummingbird/ml/supported.py:27]: Can you please also add `RandomizedSearchCV` here?

Comment (scnakandala) [reply hummingbird/ml/supported.py:27]: done.

**Comment by interesaaat** [comment]
Thanks @scnakandala!
